producer/src/test/kotlin/streams/integrations/KafkaEventRouterIT.kt

```diff
@@ -166,6 +182,17 @@ class KafkaEventRouterIT {
         consumer.close()
     }
 
+    @Test
+    fun testCantPublishNull() {
```

Comment:
@conker84  in a comment said that we should document this behavior, so now we throw an Exception and verify that is the case.

I wonder if we should allow publishing null though....I hear some people use this in kafka as a "tombstone record", but not sure if that applies here.  I'm new to this repo so I'm sticking with current behavior until we have a different requirement.